src/cli/node/json_rpc_client.rs

```diff
@@ -247,56 +247,53 @@ pub fn get_utxo_info(
     Ok(())
 }
 
+#[allow(clippy::cast_precision_loss)]
 pub fn get_reputation(
     addr: SocketAddr,
-    pkh: Option<PublicKeyHash>,
+    opt_pkh: Option<PublicKeyHash>,
     all: bool,
 ) -> Result<(), failure::Error> {
     let mut stream = start_client(addr)?;
 
-    if all {
-        let request = r#"{"jsonrpc": "2.0","method": "getReputationAll", "id": "1"}"#;
-        let response = send_request(&mut stream, &request)?;
-        let rep_map = parse_response::<HashMap<PublicKeyHash, (Reputation, bool)>>(&response)?;
-        println!("Total Reputation: {{");
-        for (pkh, (rep, active)) in rep_map
-            .into_iter()
-            .sorted_by_key(|&(_, (r, _))| std::cmp::Reverse(r))
-        {
-            let active = if active { 'A' } else { ' ' };
-            println!("    [{}] {}: {}", active, pkh, rep.0);
-        }
-        println!("}}");
-        return Ok(());
-    }
-
-    let pkh = match pkh {
-        Some(pkh) => pkh,
-        None => {
-            log::info!("No pkh specified, will default to node pkh");
-            let request = r#"{"jsonrpc": "2.0","method": "getPkh", "id": "1"}"#;
-            let response = send_request(&mut stream, &request)?;
-            let node_pkh = parse_response::<PublicKeyHash>(&response)?;
-            log::info!("Node pkh: {}", node_pkh);
-
-            node_pkh
+    let pkh = if all {
+        PublicKeyHash::default()
+    } else {
+        match opt_pkh {
+            Some(pkh) => pkh,
+            None => {
+                log::info!("No pkh specified, will default to node pkh");
+                let request = r#"{"jsonrpc": "2.0","method": "getPkh", "id": "1"}"#;
+                let response = send_request(&mut stream, &request)?;
+                let node_pkh = parse_response::<PublicKeyHash>(&response)?;
+                log::info!("Node pkh: {}", node_pkh);
+
+                node_pkh
+            }
         }
     };
 
+    let params = (pkh, all);
     let request = format!(
         r#"{{"jsonrpc": "2.0","method": "getReputation", "params": [{}], "id": "1"}}"#,
```

Comment:
Here the request will be formatted as one of the two options, depending on the `--all` flag:

```json
{"jsonrpc": "2.0","method": "getReputation", "params": [["twit1fulan0j78dmdfa3gx0779kszj5sc5j35feljv5",false]], "id": "1"}
{"jsonrpc": "2.0","method": "getReputation", "params": [["twit1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqkz4ay",true]], "id": "1"}
```

As you can see this is pretty strange, I think I prefer the old format, even if it means adding back the `getReputationAll` JSON-RPC method:

```json
{"jsonrpc": "2.0","method": "getReputation", "params": ["twit1fulan0j78dmdfa3gx0779kszj5sc5j35feljv5"], "id": "1"}
{"jsonrpc": "2.0","method": "getReputationAll", "id": "1"}
```